pkg/services/serviceaccounts/manager/service.go

```diff
@@ -125,10 +144,62 @@ func (sa *ServiceAccountsService) CreateServiceAccount(ctx context.Context, orgI
 	return sa.store.CreateServiceAccount(ctx, orgID, saForm)
 }
 
-func (sa *ServiceAccountsService) DeleteServiceAccount(ctx context.Context, orgID, serviceAccountID int64) error {
-	return sa.store.DeleteServiceAccount(ctx, orgID, serviceAccountID)
+func (sa *ServiceAccountsService) RetrieveServiceAccount(ctx context.Context, orgID int64, serviceAccountID int64) (*serviceaccounts.ServiceAccountProfileDTO, error) {
+	return sa.store.RetrieveServiceAccount(ctx, orgID, serviceAccountID)
 }
 
 func (sa *ServiceAccountsService) RetrieveServiceAccountIdByName(ctx context.Context, orgID int64, name string) (int64, error) {
 	return sa.store.RetrieveServiceAccountIdByName(ctx, orgID, name)
 }
+
+func (sa *ServiceAccountsService) DeleteServiceAccount(ctx context.Context, orgID, serviceAccountID int64) error {
+	return sa.store.DeleteServiceAccount(ctx, orgID, serviceAccountID)
+}
+
+func (sa *ServiceAccountsService) UpdateServiceAccount(ctx context.Context, orgID int64, serviceAccountID int64, saForm *serviceaccounts.UpdateServiceAccountForm) (*serviceaccounts.ServiceAccountProfileDTO, error) {
+	return sa.store.UpdateServiceAccount(ctx, orgID, serviceAccountID, saForm)
+}
+
+func (sa *ServiceAccountsService) SearchOrgServiceAccounts(
+	ctx context.Context,
+	orgID int64,
+	query string,
+	filter serviceaccounts.ServiceAccountFilter,
+	page,
+	limit int,
+	signedInUser *user.SignedInUser,
```

Comment:
Would convert this to a struct query type. (`SearchOrgServiceAccountsQuery`?)

Reply:
refactored in commit: a7b139c3a154c14fb17d70083de9a6a22b397982